net/libfab/libfab.c

```diff
@@ -360,242 +352,51 @@ M0_INTERNAL void m0_net_libfab_fini(void)
-		memcpy(port, at, (strlen(at)+1));
-	}
-	M0_ASSERT(nodeSize >= (at - ep_name));
-	memcpy(node, ep_name, ((at - ep_name)-1));
-	return 0;
-}
-
-/**
- * Used to decode the ip and port from the given end point
- * ep_name : endpoint address from domain
- * node    : copy ip address from ep_name
- * port    : copy port number from ep_name
- * Example of ep_name IPV4 192.168.0.1:4235
- *                    IPV6 [4002:db1::1]:4235
- */
-static int libfab_ep_addr_decode_native(const char *ep_name, char *node,
-					size_t node_size, char *port,
-					size_t port_size)
-{
-	const char *name;
-	const char *cp;
-	size_t      n;
-
-	M0_PRE(ep_name != NULL);
-
-	M0_ENTRY("ep_name=%s", ep_name);
-
-	name = ep_name;
-
-	if (name[0] == '[') {
-		/* IPV6 pattern */
-		cp = strchr(name, ']');
-		if (cp == NULL)
-			return M0_ERR(-EINVAL);
-
-		name++;
-		n = cp - name;
-		if (n == 0)
-			return M0_ERR(-EINVAL);
-		cp++;
-		if (*cp != ':')
-			return M0_ERR(-EINVAL);
-		cp++;
-	} else {
-		/* IPV4 pattern */
-		cp = strchr(name, ':');
-		if (cp == NULL)
-			return M0_ERR(-EINVAL);
-
-		n = cp - name;
-		if (n == 0)
-			return M0_ERR(-EINVAL);
-
-		++cp;
-	}
-
-	if (n + 1 > node_size || strlen(cp) + 1 > port_size)
-		return M0_ERR(-EINVAL);
-
-	node[0] = '\0';
-	strncat(node, name, n);
-
-	port[0] = '\0';
-	strncat(port, cp, port_size-1);
-
-	return M0_RC(0);
+	if (addr->nip_format == M0_NET_IP_LNET_FORMAT)
+		inet_ntop(AF_INET, &addr->nip_ip_n.sn[0], ip,
+			  LIBFAB_ADDR_LEN_MAX);
+	else if (addr->nip_fmt_pvt.ia.nia_family == M0_NET_IP_AF_INET)
+		inet_ntop(AF_INET, &addr->nip_ip_n.sn[0], ip,
+			  LIBFAB_ADDR_LEN_MAX);
+	else if (addr->nip_fmt_pvt.ia.nia_family == M0_NET_IP_AF_INET6)
+		inet_ntop(AF_INET6, &addr->nip_ip_n.ln[0], ip,
+			  LIBFAB_ADDR_LEN_MAX);
+	else
+		M0_LOG(M0_ERROR, "Family is not supported.");
 }
-
 /**
  * Parses network address.
  *
  * The following address formats are supported:
  *
- *     - lnet compatible, see nlx_core_ep_addr_decode():
- *
- *           nid:pid:portal:tmid
- *
- *       for example: "10.0.2.15@tcp:12345:34:123" or
- *       "192.168.96.128@tcp1:12345:31:*"
- *
- *     - sock format, see socket(2):
- *           family:type:ipaddr[@port]
+ *     - The lnet compatible format is of type
+ *        <nid>@<type>:<pid>:<portal>:<tmid>
+ *       for example: "10.0.2.15@tcp:12345:34:123",
+ *                    "192.168.96.128@tcp:12345:31:*",
+ *                    see nlx_core_ep_addr_decode()
  *
- *     - libfab compatible format
- *       for example IPV4 libfab:192.168.0.1:4235
- *                   IPV6 libfab:[4002:db1::1]:4235
+ *     - The inet address format is of type
+ *         <family>:<type>:<ipaddr/hostname_FQDN>@<port>
+ *        for example: "inet:tcp:127.0.0.1@3000",
+ *                     "inet:stream:lanl.gov@23",
+ *                     "inet6:dgram:FE80::0202:B3FF:FE1E:8329@6663"
  *
  */
 static int libfab_ep_addr_decode(struct m0_fab__ep *ep, const char *name,
 				 struct m0_fab__ndom *fnd)
```

Comment:
This param fnd seems to be unused and can be removed

Reply:
Removed.